export: add memory monitoring for EXPORT INTO PARQUET

jayshrivastava opened the pull request:
This change introduces a simple memory monitoring scheme
for `EXPORT INTO PARQUET`. For each datum, we now request
an allocation of 3 times the size of the datum from
the parent memory monitor. Any failure to allocate this
will return an error instead of OOMing the node.

This multiplication factor of 3 can be changed via the
cluster setting export.event_memory_multiplier.

Informs: https://github.com/cockroachdb/cockroach/issues/103317
Release note: None
Epic: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/105064)
<!-- Reviewable:end -->

jayshrivastava committed:
export: add memory monitoring for `EXPORT INTO PARQUET`

This change introduces a simple memory monitoring scheme
for `EXPORT INTO PARQUET`. For each datum, we now request
an allocation of 3 times the size of the datum from
the parent memory monitor. Any failure to allocate this
will return an error instead of OOMing the node.

This multiplication factor of 3 can be changed via the
cluster setting export.event_memory_multiplier.

Informs: https://github.com/cockroachdb/cockroach/issues/103317
Release note: None
Epic: None

jayshrivastava commented:
bors r=miretskiy

jayshrivastava commented:
blathers backport 23.1

jayshrivastava commented:
blathers backport 22.2
